internal/docker/deployer.go

```diff
@@ -391,6 +348,39 @@ func deployImage(
 	return d, nil
 }
 
+// Picks two free ports on localhost. Does not reserve them in any way.
+// The returned ports must be used before the next call to `allocateHostPorts`,
+// otherwise the same pair of ports may be returned.
```

Comment:
We cannot guarantee that as homeservers are deployed in parallel.

Reply:
That's a good point, I didn't realise that we ran tests in parallel at the time.